src/services/user-secret-creator.js

```diff
@@ -0,0 +1,20 @@
+const base32Encode = require('base32-encode');
+const xor = require('bitwise-xor');
+
+// NOTICE: This service combines the 2FA secret stored on the forest server to the local secret
+//         salt. This guarantees that only the owner of the server and the concerned end user can
+//         know the final key.
+//         This is done by using a bitwise exclusive or operation, which guarantees the key to stay
+//         unique, so it is impossible for two users to have the same key.
```

Comment:
Ok!

(Considering that the 2FSecretSalt is the same for all users of a project and that each user secret is supposed to be different accross the list of users of a project.)